server/services/lists/caCaseloadService.ts

```diff
@@ -59,137 +63,186 @@ export type ManagedCase = {
+    user: User,
+    prisonCaseload: string[],
+    searchString?: string
+  ): Promise<CaCaseLoad> {
+    const licences = await this.licenceService.getPostReleaseLicencesForOmu(user, prisonCaseload)
+
+    if (!licences.length) {
+      return { cases: [], showAttentionNeededTab: false }
+    }
+
+    const formattedLicences = this.formatReleasedLicences(licences)
+    const cases = await this.mapCasesToComs(formattedLicences)
+
+    return this.buildCaseload(cases, searchString, 'probation')
   }
 
-  public isProbationCase(managedCase: ManagedCase): boolean {
-    return this.hasAnyStatusOf(PROBATION_VIEW_STATUSES, managedCase)
+  private async filterAndFormatExistingLicences(licences: LicenceSummary[], user: User): Promise<CaCase[]> {
+    if (!licences.length) {
+      return []
+    }
+
+    const licenceNomisIds = licences.map(l => l.nomisId)
+    const prisonersWithLicences = await this.licenceService.searchPrisonersByNomsIds(licenceNomisIds, user)
+    const nomisEnrichedLicences = this.enrichWithNomisData(licences, prisonersWithLicences)
+    return this.filterExistingLicencesForEligibility(nomisEnrichedLicences)
   }
 
-  public async getOmuCaseload(user: User, prisonCaseload: string[]): Promise<ManagedCase[]> {
-    // Get cases with a licence in ACTIVE, APPROVED, SUBMITTED, IN_PROGRESS or VARIATION_IN_* state
-    const casesWithLicences = this.licenceService
-      .getLicencesForOmu(user, prisonCaseload)
-      .then(licences => this.mapLicencesToOffenders(licences))
+  private async findAndFormatNotStartedLicences(
+    licences: LicenceSummary[],
+    prisonCaseload: string[],
+    user: User
+  ): Promise<CaCase[]> {
+    const licenceNomisIds = licences.map(l => l.nomisId)
+    const prisonersApproachingRelease = await this.getPrisonersApproachingRelease(user, prisonCaseload)
 
-    // Get cases due for release soon which do not have a submitted licence
-    const today = startOfDay(new Date())
-    const todayPlusFourWeeks = endOfDay(add(new Date(), { weeks: 4 }))
-    const casesPendingLicence = this.licenceService
-      .searchPrisonersByReleaseDate(today, todayPlusFourWeeks, prisonCaseload, user)
-      .then(caseload => this.pairNomisRecordsWithDelius(caseload))
-      .then(caseload => this.filterOffendersEligibleForLicence(caseload, user))
-      .then(caseload => this.mapOffendersToLicences(caseload, user))
-      .then(caseload => this.buildCreateCaseload(caseload))
-      .then(caseload => {
-        return caseload
-          .filter(c => !c.licences.find(l => l.status === LicenceStatus.TIMED_OUT && l.id))
-          .filter(c =>
-            [
-              LicenceStatus.NOT_STARTED,
-              LicenceStatus.TIMED_OUT,
-              LicenceStatus.NOT_IN_PILOT,
-              LicenceStatus.OOS_RECALL,
-              LicenceStatus.OOS_BOTUS,
-            ].some(status => c.licences.find(l => l.status === status))
-          )
-      })
+    if (!prisonersApproachingRelease.length) {
+      return []
+    }
```

Comment:
**quibble:(non-blocking)** There's a lot of defensive coding around checking whether arrays have lengths..

Here we're just avoiding filtering over an empty array which would then be filtered out anyway.
Generally good to avoid these "edges" ([obligatory Michael Feathers link again](https://www.youtube.com/watch?v=vbLyfh6QdI0)) and use empty arrays as sensible null values that can pass through subsequent functions. 

There obviously are places where you need to handle these, we don't really want to do api calls to retrieve no information but these can generally be implementation details of the clients (and probably are in most if not all places)

Reply:
Great point, not sure why I originally thought it was necessary to put this one in
If I end up having to make any other changes I'll remove this and do a scan through to remove any others that don't really need to be there